test_server_q3.py expects newline when reading thought

michaldeutch opened the pull request:
from Exercise 1:Introduction-

For example, if user 1 also thought I'm sleepy at the same time, the file /tmp/data/1/2019-10-25_15-12-05.txt **should contain:
I'm hungry
I'm sleepy**

So it seems that after each thought a newline is expected.

michaldeutch1 committed:
added rstrip() when thought is read from file

dan-gittik commented:
Actually, `test_server_q3.py` doesn't expect a trailing newline -- the `.rstrip()` you added was to your output, not to the expected output.
Anyway, the test is correct: there shouldn't be a trailing newline, even when there are multiple thoughts in the same file.

michaldeutch commented:
Hi, thank you.
I understood that I added rstrip() to my output, since read_text() returns the content of the data file as a string, and I added a trailing newline for each thought.

If the tests are supposed to pass as you suggest, then how each thought is supposed to be seperated? test_race_condition calls read_text().splitlines() and that suggests each thought is seperated with a newline.

I would be great if you can clear this up.
